### ISSUE TITLE
[Bug] Samsung S21 Crash - storage permissions 

### ISSUE THREAD

**Issue by Waterfall2022** [ankidroid/Anki-Android#12156]
###### Reproduction Steps

1. Open fresh install of app
2. Prompted to enable permissions 
3. As soon as permission given to phone storage phone crashed. 

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
Could you provide this?

> Debug Info
> Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

Then we could attempt to find it in our crash database, we really need the stack trace in order to understand this

**Comment by BrayanDSO** [comment]
If this is constant, and the app crashes immediately after opening it, they can't go to the "Copy debug info" button.

**Comment by Waterfall2022** [comment]
I've submitted the crash report upon startup.

I inputted the comment "yellowstone" on 1222 EST

**Comment by mikehardy** [comment]
Sadly, I don't see a report that looks like a match at all when I check our crash collection server for Samsung Galaxy S21+ devices. Can you make sure you are trying the most up to date alpha from here: https://github.com/ankidroid/Anki-Android/releases and can you confirm any other details about the device (it appears some are on Android 11, some are on Android 12? Maybe if we knew the device language + timezone we could really match it?)

Without the precise installation ID it is sometimes hard to say if we have the crash dump or not, because we don't collect anything identifiable on purpose :sweat_smile:

**Comment by Waterfall2022** [comment]
Just did it with the latest version.
Android 12. Timezone EST?

Thanks again for your help!!

**Comment by mikehardy** [comment]
With apologies, I simply cannot find it. Without that stack trace I'm not sure how to move forward on this one :disappointed: 

![image](https://user-images.githubusercontent.com/782704/186508305-6e1414be-41d4-468a-a3e1-1bbae5534335.png)

**Comment by BrayanDSO** [comment]
Hmmm, I think that adding a "Copy debug info" to that "Send error report"/"Asking permission to send report" dialog may help for similar issues (instant crash) in the future.

**Comment by mikehardy** [comment]
Perhaps yes, however at this point I think it's a bigger deal - the report isn't going out *at all*. Your suggestion would be an improvement for startup crashes in general though

**Comment by Waterfall2022** [comment]
Maybe one of those is mine 👀😬

Anything I can do on my end to generate a local crash report that I can
copy and paste to yall?

I tried much older releases too but they dont work either.


I really do appreciate the help and the prompt responses!

In the meantime I've figured out to use anki web. It's not ideal but itll
do!

**Comment by mikehardy** [comment]
All I can think of is that if you remove storage permission via the system preferences / apps panel, then when AnkiDroid starts up maybe it won't crash immediately, and that will let it live long enough for the crashes to be sent? I'm just grasping at straws though unfortunately. I really don't know what else to think of. If your collection is in sync via AnkiWeb and you can use it that is great. Pretty soon we'll have a new storage mechanism (that doesn't require the permission) then this whole issue will likely go away - follow #5304

**Comment by Waterfall2022** [comment]
Disabled it but when I relaunch the app it prompts me to enable permission.
If I try to continue phone permission manager setting pop-ups to enable
permissions. So im stuck :3

**Comment by escape0707** [comment]
> Perhaps yes, however at this point I think it's a bigger deal - the report isn't going out _at all_. Your suggestion would be an improvement for startup crashes in general though

I was always wondering if there is a local copy of crash report so that users can review them, too? Like before submitting or after?

**Comment by mikehardy** [comment]
Indeed there is! But it's hard to get to.

Android is Unix, which means "everything is a file", so the whole phone is just one big filesystem with a bunch of files. Just to start conceptually.

There are permissions though, so some stuff you can't see, and it's hard to access the files sometimes if you don't have a good app for it (I use X-Plore) or you don't have "android debugging bridge" (adb) and a laptop and developer mode set on the phone and skills at using `adb shell` to go on the phone and sniff around. (ou may, if so awesome, but most don't)

On a normal phone that is not rooted I don't think you can see it at all, but if you could, it would be in the path `/data/data/com.ichi2.anki/` then I think a folder called ACRA. That's where they are when I'm testing the crash reporting infrastructure on emulators (which are rooted) anyway. I can't see them with X-Plore on my real phone just now, that's considered private data and the system doesn't grant read to it

**Comment by escape0707** [comment]
Oh, I see that I can't view them even with adb.

Maybe this could be improved once we move to new scoped storage scheme? Like put the crash log into external app-specific storage?

```
bluejay:/storage/emulated/0/Android/data/com.ichi2.anki $ ls -R
.:
cache  files

./cache:

./files:

```

**Comment by Waterfall2022** [comment]
Hey jw.... how did you guys started in this field. I think it's really cool
that you guys built and support a useful product.

**Comment by david-allison** [comment]
I believe this is fixed in #12637

**Comment by mikehardy** [comment]
We are working to get fresh releases in the app store, in the meantime access to fixes is restricted to alpha releases you download from our releases page. I have not packed up the current set of changes and released an alpha yet though, you'll find this change in alphas 2.16alpha87 (not released yet) and higher here https://github.com/ankidroid/Anki-Android/releases
